app/controllers/api/v1/beers_controller.rb

```diff
@@ -2,43 +2,54 @@ module Api
   module V1
     class BeersController < Api::V1::BaseController
       before_action :authenticate_with_token, only: [:create]
+      before_action :find_or_initialize_beer, only: [:create]
       respond_to :json
 
       def index
-        respond_with beers: Beer.search(search_params)
+        respond_with beers: filter_beers
       end
 
       def create
-        beer = find_or_initialize_beer
-
-        if beer.save
-          Rating.create_with_relationships(@user, beer, beer_params[:rating])
-          respond_with beer, location: nil
+        if @beer.save
+          Rating.create_with_relationships(@user, @beer, beer_params[:rating])
+          respond_with @beer, location: nil
         else
-          errors = { errors: beer.errors }
+          errors = { errors: @beer.errors }
           respond_with errors, location: nil, status: 422
         end
       end
 
       private
 
-        def find_or_initialize_beer
-          beer = Beer.find_or_initialize_by(name: beer_params[:name].to_s.downcase,
-                                            beer_type: beer_params[:beer_type].to_s.downcase,
-                                            brand: beer_params[:brand].to_s.downcase)
+      def filter_beers
+        Beer.current_beers(search_params[:current_beers],
+                           search_params[:token])
+            .beer_type(search_params[:type])
+            .beer_name(search_params[:text])
+            .sort_by_criterion(search_params[:sort])
+            .current_page(search_params[:page])
+            .beer_order
```

Comment:
So this looks better, but you're using too may scopes and not using `params.permit` or `params.require`. 

```
def permitted_params
  params.permit(:type, :text)
end
```
 would allow you to write `Beer.where(permitted_params)` then apply custom scopes.

Something like this may work better because it gives you optional, flexible queries to run on it.
```
relation = Beer.scope_by_user_id(params[:user_id])
relation = relation.where(permitted_params)
relation = relation.sort_scope(params[:sort]) if params.key? :sort
relation = relation.limit(params[:limit]) if params.key? :limit
relation = relation.offset(params[:offset]) if params.key? :offset
relation = relation.scope_by_like_name() ... # where("name ILIKE ?" params[:name_query]) # ILIKE will be case insensitive
... (add more as needed)
```